crypto/x509/x509_vfy.c

```diff
@@ -1737,8 +1741,8 @@ static int internal_verify(X509_STORE_CTX *ctx)
         goto check_cert_time;
     }
 
-    /* Make sure that the root of the chain is trusted */
-    CB_FAIL_IF(!X509_check_trust(xi, ctx->param->trust, ctx->param->flags),
+    /* Make sure that the root of the chain is directly trusted */
+    CB_FAIL_IF(!directly_trusted(ctx, xi),
```

Comment:
Sorry, on balance I think this PR should be abandoned.  The `internal_verify()` function is not the right place to second-guess the trust-path construction.  Its job is solely to check chain integrity (signatures and dates).

Reply:
As meanwhile discussed at https://github.com/openssl/openssl/issues/13780#issuecomment-759971511 the `internal_verify()` function may not be the right/best place to add the intended check, so I've disabled it there for now.
Yet where else to place it then?

Reply:
> As meanwhile discussed at [#13780 (comment)](https://github.com/openssl/openssl/issues/13780#issuecomment-759971511) the `internal_verify()` function is not the right/best place to add the intended check, so I've disabled it there for now.
> Yet where else to place it then?

Nowhere that comes to mind.  Hence the suggestion to drop this PR.

Reply:
I meanwhile think that the check is best done just before calling `internal_verify()`, in `verify_chain()`.

Reply:
I don't see how that solves my fundamental objection that the context for making decisions is no longer available, and we just need to make sure that the build_chain code is correct.